src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/AutoInsert/OnAutoInsertEndpoint.cs

```diff
@@ -84,36 +83,20 @@ public void ApplyCapabilities(VSInternalServerCapabilities serverCapabilities, V
 
         var character = request.Character;
 
-        using var applicableProviders = new PooledArrayBuilder<IOnAutoInsertProvider>();
-        foreach (var provider in _onAutoInsertProviders)
-        {
-            if (provider.TriggerCharacter == character)
-            {
-                applicableProviders.Add(provider);
-            }
-        }
-
-        if (applicableProviders.Count == 0)
-        {
-            // There's currently a bug in the LSP platform where other language clients OnAutoInsert trigger characters influence every language clients trigger characters.
-            // To combat this we need to preemptively return so we don't try having our providers handle characters that they can't.
-            return null;
-        }
+        var insertTextEdit = _autoInsertService.TryResolveInsertion(
+            codeDocument,
+            request.Position,
+            character,
+            _optionsMonitor.CurrentValue.AutoClosingTags
+        );
```

Comment:
nit: Please move `);` to the following line.

Reply:
Did you mean to the preceding?